securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/api/ConsentDecisionApiController.java

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.api;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.forgerock.securebanking.openbanking.uk.common.claim.Claims;
+import com.forgerock.securebanking.openbanking.uk.common.claim.JwsClaimsUtils;
+import com.forgerock.securebanking.openbanking.uk.error.OBErrorException;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.RedirectionAction;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.decision.ConsentDecision;
+import com.forgerock.securebanking.openbanking.uk.rcs.common.RCSConstants;
+import com.forgerock.securebanking.openbanking.uk.rcs.service.decision.ConsentDecisionService;
+import com.forgerock.securebanking.openbanking.uk.rcs.service.decision.ConsentDecisionServiceDelegate;
+import com.forgerock.securebanking.openbanking.uk.rcs.validator.ConsentDecisionValidator;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import com.nimbusds.jwt.SignedJWT;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+import java.text.ParseException;
+
+import static com.forgerock.securebanking.openbanking.uk.common.api.meta.OBConstants.IdTokenClaim.INTENT_ID;
+import static com.forgerock.securebanking.openbanking.uk.error.OBRIErrorType.RCS_CONSENT_DECISION_EMPTY;
+import static com.forgerock.securebanking.openbanking.uk.error.OBRIErrorType.RCS_CONSENT_REQUEST_FORMAT;
+import static com.forgerock.securebanking.openbanking.uk.rcs.common.RCSConstants.Claims.CLIENT_ID;
+import static com.forgerock.securebanking.openbanking.uk.rcs.util.ConsentDecisionDeserializer.deserializeConsentDecision;
+
+@Controller
+@Slf4j
+public class ConsentDecisionApiController implements RCSConsentDecisionApi {
+
+    private final ConsentDecisionResponseHandler consentDecisionResponseHandler;
+    private final ObjectMapper objectMapper;
+    private final ConsentDecisionServiceDelegate consentServiceDelegate;
+    private final ConsentDecisionValidator consentDecisionValidator;
+
+    public ConsentDecisionApiController(ConsentDecisionResponseHandler consentDecisionResponseHandler,
+                                        ObjectMapper objectMapper,
+                                        ConsentDecisionServiceDelegate consentServiceDelegate,
+                                        ConsentDecisionValidator consentDecisionValidator) {
+        this.consentDecisionResponseHandler = consentDecisionResponseHandler;
+        this.objectMapper = objectMapper;
+        this.consentServiceDelegate = consentServiceDelegate;
+        this.consentDecisionValidator = consentDecisionValidator;
+    }
+
+    @Override
+    public ResponseEntity<RedirectionAction> submitConsentDecision(String consentDecisionSerialised,
+                                                                   String ssoToken) throws OBErrorException {
```

Comment:
A re-write of [RCSConsentDecisionApiController](https://github.com/OpenBankingToolkit/openbanking-aspsp/blob/62ebb09658028a347c362ddd93610b1966b6ae03/forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-rcs/forgerock-openbanking-uk-aspsp-rs-rcs-server/src/main/java/com/forgerock/openbanking/aspsp/rs/rcs/api/rcs/decisions/RCSConsentDecisionApiController.java)

By separating concerns into different classes (i.e. a Single Responsibility approach), this method is now 31 lines instead of 115.